inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+        ngraph::get_opset4(), ngraph::get_opset5()};
+
+    // return the oldest opset name where node type is present
+    for (int idx = 0; idx < opsets.size(); idx++) {
+        int number = idx + 1;
+        if (opsets[idx].get().contains_op_type(n)) {
+            return "opset" + std::to_string(number);
+        }
+    }
+    return "experimental";
+}
+
+// Here operation type names are translated from ngraph convention to IR
+// convention. Most of them are the same, but there are exceptions, e.g
+// Constant (ngraph name) and Const (IR name). If there will be more
+// discrepancies discoverd, translations needs to be added here.
+std::string get_type_name(const ngraph::Node* n) {
+    std::string name = n->get_type_name();
+    NGRAPH_CHECK(name != "GenericIE", "Type of ", n->get_name(), " (",
+                 n->get_type_name(), ") ", "is not supported.");
+
+    const std::map<std::string, std::string> translator = {
+        {"Constant", "Const"}};
+    if (translator.count(name) > 0) {
+        name = translator.at(name);
+    }
+    return name;
+}
+
+std::string get_output_precision_name(ngraph::Output<Node>& o) {
+    auto elem_type = o.get_element_type();
+    switch (elem_type) {
+    case ::ngraph::element::Type_t::undefined:
+        return "UNSPECIFIED";
+    case ::ngraph::element::Type_t::f16:
+        return "FP16";
+    case ::ngraph::element::Type_t::f32:
+        return "FP32";
+    case ::ngraph::element::Type_t::bf16:
+        return "BF16";
+    case ::ngraph::element::Type_t::i8:
+        return "I8";
+    case ::ngraph::element::Type_t::i16:
+        return "I16";
+    case ::ngraph::element::Type_t::i32:
+        return "I32";
+    case ::ngraph::element::Type_t::i64:
+        return "I64";
+    case ::ngraph::element::Type_t::u8:
+        return "U8";
+    case ::ngraph::element::Type_t::u16:
+        return "U16";
+    case ::ngraph::element::Type_t::u32:
+        return "U32";
+    case ::ngraph::element::Type_t::u64:
+        return "U64";
+    case ::ngraph::element::Type_t::u1:
+        return "BIN";
+    case ::ngraph::element::Type_t::boolean:
+        return "BOOL";
+    default:
+        NGRAPH_CHECK(false, "Incorrect precision: ", elem_type.get_type_name());
+        return "";
+    }
+}
+
+std::string generate_unique_name(
+    const std::unordered_set<std::string>& unique_names, std::string base_name,
+    int suffix) {
+    std::string new_name = base_name + std::to_string(suffix);
+    if (unique_names.find(new_name) == unique_names.end()) {
+        return new_name;
+    } else {
+        suffix++;
+        return generate_unique_name(unique_names, base_name, suffix);
+    }
+}
+
+// TODO: remove when CNNNetwork will be supporting not-unique names
+std::string get_node_unique_name(std::unordered_set<std::string>& unique_names,
+                                 const ngraph::Node* n) {
+    std::string name = n->get_friendly_name();
+    if (unique_names.find(name) != unique_names.end()) {
```

Comment:
@ilyachur I think we never have two nodes with the same friendly name in ngraph. Do we really need this check?

Reply:
It came up in functional tests when function was created on ONNX Importer path - not unique names were present, hence after consultation with Ilya I added this code to ensure uniqueness.

Reply:
We can get the same friendly names from ONNX model